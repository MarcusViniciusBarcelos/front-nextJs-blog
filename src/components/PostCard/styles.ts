import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6);
      margin-bottom: ${theme.spacings.small};
    }

    h2 {
      transition: all 300ms ease-in-out;
    }

    &:hover h2 {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
      transform: scale(1.03);
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: all 300ms ease-in-out;
`;

export const Excerpt = styled.p`
  max-width: 100%;
`;
