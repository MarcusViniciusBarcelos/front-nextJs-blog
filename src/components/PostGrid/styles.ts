import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin: 0 auto;
  `}
`;

export const NotFound = styled.p`
  ${({ theme }) => css`
    padding: ${theme.spacings.large};
    text-align: center;
    font-size: ${theme.font.sizes.small};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    margin: 0 auto;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};
    padding: 0 ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;
