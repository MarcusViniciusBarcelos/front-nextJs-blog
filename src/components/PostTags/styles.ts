import styled, { css } from 'styled-components';

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} auto;
    max-width: ${theme.sizes.content};
    padding: 0 ${theme.spacings.large} 0;

    span {
      margin-left: 0.5rem;
      background: rgba(200, 200, 200, 0.4);
      padding: 0.4rem 0.4rem;
      border-radius: 0.5rem;
      font-size: 1.5rem;
      transition: all 0.3s ease-in-out;
      cursor: pointer;

      &:hover {
        background: rgba(200, 200, 200, 0.6);
      }
    }

    span::before {
      content: ' ';
    }

    span::after {
      content: ' ';
    }

    a {
      transition: all 0.3s ease-in-out;
      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
