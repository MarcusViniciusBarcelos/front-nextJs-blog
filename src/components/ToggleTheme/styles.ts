import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
`;

export const label = styled.label`
  ${({ theme }) => css`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    font-size: 0;
    overflow: hidden;
    color: rgba(0, 0, 0, 0);
  `}
`;

export const input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background-color: ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
    }

    &:focus + ${Slider} {
      border: 1px solid ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
    }

    &:checked + ${Slider}:before {
      transform: translateX(2rem);
      transition: all 300ms ease-in-out;
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: ${theme.colors.darkerGray};
    transition: all 300ms ease-in-out;
    border-radius: 2rem;
    border: 1px solid ${theme.colors.darkerGray};

    &::before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background-color: ${theme.colors.white};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
