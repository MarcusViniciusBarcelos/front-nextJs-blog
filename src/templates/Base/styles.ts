import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    svg.search-ok-icon {
      width: 3rem;
      height: 3rem;
      margin-left: 0.5rem;
    }

    svg.search-cancel-icon {
      width: 3rem;
      height: 3rem;
      margin-left: 0.5rem;
      color: ${theme.colors.secondary};
    }
  `}
`;

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xhuge};
    padding: 0 ${theme.spacings.large};
  `}
`;

export const SearchContainer = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xlarge};
    padding: 0 ${theme.spacings.large};
    max-width: 120rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export const SearchInput = styled.input`
  ${({ theme }) => css`
    padding: 0.5rem ${theme.spacings.small};

    &:disabled {
      outline: 0.1rem solid ${theme.colors.secondary};
    }
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.large};
  `}
`;

export const FooterContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
