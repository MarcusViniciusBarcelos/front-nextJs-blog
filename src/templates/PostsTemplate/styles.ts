import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    padding: ${theme.spacings.small} ${theme.spacings.large};
    border: none;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    &:disabled {
      background: ${theme.colors.darkerGray};
    }
  `}
`;
