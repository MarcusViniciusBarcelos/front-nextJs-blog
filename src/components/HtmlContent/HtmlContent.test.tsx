import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { HtmlContent } from '.';
import '@testing-library/jest-dom';

describe('<HtmlContent />', () => {
  it('should render a text', () => {
    renderTheme(<HtmlContent html={'Children'} />);
    expect(screen.getByText('Children')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<HtmlContent html={'Children'} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
