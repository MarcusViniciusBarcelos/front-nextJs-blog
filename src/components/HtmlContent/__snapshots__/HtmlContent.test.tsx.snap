// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlContent /> should match snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
}

.c0 p {
  margin: 1.6rem 0;
}

<div
  class="c0"
>
  Children
</div>
`;
